<launch>

  <arg name="GUI" default="true" />
  <arg name="PROCESS" default="true" />
  <arg name="USE_PASSTHROUGH" default="false" />

  <group ns="/camera">
    <group ns="color">
      <group if="$(arg USE_PASSTHROUGH)">
        <node name="image_rect_color_passthrough"
              pkg="nodelet" type="nodelet"
              args="standalone jsk_topic_tools/Passthrough" if="$(arg PROCESS)">
          <remap from="~input" to="image_rect_color" />
        </node>
      </group>
      <group unless="$(arg USE_PASSTHROUGH)">
        <node name="image_rect_color_passthrough"
              pkg="nodelet" type="nodelet"
              args="standalone jsk_topic_tools/LightweightThrottle" if="$(arg PROCESS)">
          <remap from="~input" to="image_rect_color" />
          <rosparam subst_value="true">
            update_rate: 1
          </rosparam>
        </node>
      </group>
    </group>
    <node name="mask_rcnn_instance_segmentation"
          pkg="morefusion_panda_ycb_video" type="mask_rcnn_instance_segmentation.py"
          clear_params="true" output="screen" if="$(arg PROCESS)">
      <remap from="~input" to="color/image_rect_color_passthrough/output" />
    </node>
    <node name="draw_instance_segmentation"
          pkg="morefusion_panda_ycb_video" type="draw_instance_segmentation.py"
          clear_params="true">
      <remap from="~input/rgb" to="color/image_rect_color_passthrough/output" />
      <remap from="~input/label_ins" to="mask_rcnn_instance_segmentation/output/label_ins" />
      <remap from="~input/class" to="mask_rcnn_instance_segmentation/output/class" />
    </node>
    <group ns="draw_instance_segmentation" if="$(arg GUI)">
      <node name="image_view"
            pkg="image_view" type="image_view"
            clear_params="true">
        <remap from="image" to="output" />
      </node>
    </group>
  </group>

</launch>
